opslevel/resources_opslevel_team_tag.go

```diff
@@ -1,159 +1,234 @@
+	Team        types.String `tfsdk:"team"`
+	TeamAlias   types.String `tfsdk:"team_alias"`
+	Value       types.String `tfsdk:"value"`
+}
+
+func NewTeamTagResourceModel(teamTag opslevel.Tag, teamIdentifier string) TeamTagResourceModel {
+	teamResourceModel := TeamTagResourceModel{
+		Key:   types.StringValue(teamTag.Key),
+		Value: types.StringValue(teamTag.Value),
+		Id:    types.StringValue(string(teamTag.Id)),
+	}
+
+	// use either the team ID or alias based on what is used in the config
+	if opslevel.IsID(teamIdentifier) {
+		teamResourceModel.Team = types.StringValue(teamIdentifier)
+	} else {
+		teamResourceModel.TeamAlias = types.StringValue(teamIdentifier)
+	}
+
+	return teamResourceModel
+}
+
+func (teamTagResource *TeamTagResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_team_tag"
+}
+
+func (teamTagResource *TeamTagResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "Team Tag Resource",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "The ID of this resource.",
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"last_updated": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+			},
+			"key": schema.StringAttribute{
+				Description: "The tag's key.",
+				Required:    true,
+				// TODO: can add validator
+			},
+			"value": schema.StringAttribute{
+				Description: "The tag's value.",
+				Required:    true,
+			},
+			"team": schema.StringAttribute{
+				Description: "The id of the team that this will be added to.",
+				Optional:    true,
+			},
+			"team_alias": schema.StringAttribute{
+				Description: "The alias of the team that this will be added to.",
+				Optional:    true,
+			},
+		},
+	}
+}
+
+func (teamTagResource *TeamTagResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var data TeamTagResourceModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	tagCreateInput := opslevel.TagCreateInput{
+		Type:  opslevel.RefOf(opslevel.TaggableResourceTeam),
+		Key:   data.Key.ValueString(),
+		Value: data.Value.ValueString(),
+	}
+
+	// use either the team ID or alias based on what is used in the config
+	var teamIdentifier string
+	if data.Team.ValueString() == "" && data.TeamAlias.ValueString() == "" {
+		resp.Diagnostics.AddError("config error", "require at least one of: 'team', 'team_alias'")
+		return
+	}
+	if data.Team.ValueString() != "" {
+		teamIdentifier = data.Team.ValueString()
+		tagCreateInput.Id = opslevel.NewID(teamIdentifier)
+	} else {
+		teamIdentifier = data.TeamAlias.ValueString()
+		tagCreateInput.Alias = &teamIdentifier
```

Comment:
```suggestion
		tagCreateInput.Alias = data.TeamAlias.ValueStringPointer()
```

Reply:
I still need to set `teamIdentifier` here, so I am not adding this suggestion